### ISSUE TITLE
Unable to split the data in fit_fast() if my countries are less then 5. 

### ISSUE THREAD

**Issue by FaisalMoosani** [microsoft/SparseSC#48]
I am facing "ValueError: Cannot have number of splits n_splits=5 greater than the number of samples: n_samples=4." issue in fit_fast() when my countries in data are less then 5. 

<img width="1092" alt="Screenshot 2023-05-04 at 22 05 51" src="https://user-images.githubusercontent.com/39930777/236318089-2f6ba6e7-6495-4bfc-87c7-2840ddcecf70.png">

**Comment by FaisalMoosani** [comment]
<img width="1092" alt="Screenshot 2023-05-04 at 22 06 23" src="https://user-images.githubusercontent.com/39930777/236318189-0c20712c-a29e-4843-81e5-6d904fb661d8.png">
<img width="1092" alt="Screenshot 2023-05-04 at 22 07 00" src="https://user-images.githubusercontent.com/39930777/236318210-a0153a4d-b421-43d1-8bdf-f456c2384d9e.png">

**Comment by jdthorpe** [comment]
I suspect that if you look at the trace, that this error actually comes from [sklearn's Kfold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html) function which is used for splitting the records in the data into folds (subsets of rows) for cross-validation and k-fold gradient descent.  The default for the number of is 5 folds (splits) is [here](https://github.com/microsoft/SparseSC/blob/acd2ad1adacff3e6925171dc0b46e328b7a0e92d/src/SparseSC/fit_fold.py#L25) and [here](https://github.com/microsoft/SparseSC/blob/acd2ad1adacff3e6925171dc0b46e328b7a0e92d/src/SparseSC/fit_fold.py#L433), and can be controlled manually using the `gradient_folds` parameter of fit().

In short, this message is just saying that you cant divide a dataset with 4 rows into 5 non-empty subsets.  

Also, as an aside, I would question the sensibility of using this methodology on such a small dataset.

**Comment by FaisalMoosani** [comment]
![image](https://user-images.githubusercontent.com/39930777/237020562-8fdb7bb0-84e0-460f-855f-771a45d84424.png)

when I try to enter gradient_folds = 4, it gives me same error.
